src/griptape_nodes/retained_mode/managers/node_manager.py

```
@@ -1194,6 +1260,10 @@ def on_alter_parameter_details_request(self, request: AlterParameterDetailsReque
             # TODO: https://github.com/griptape-ai/griptape-nodes/issues/825
             node.parameter_values[request.parameter_name] = request.default_value
 
+        # Check and handle connections if type was changed
+        if request.type is not None or request.input_types is not None or request.output_type is not None:
+            self._validate_and_break_invalid_connections(node_name, parameter, request)
```

If this has a failure, we should return a failure. Wondering if we should do this up earlier before we've actually altered the nodes.

---

ah, you mean validate and break connections earlier? Probably a good idea..

---

moved it earlier, and now returning a failure